### PR TITLE
Added resurrector; fixed provisioning

### PR DESCRIPTION
**Pull request by drbitboy**
Hi Joseph, Jared,

This is the merge of all the recent work adding resurrector.  Let me know if you want a demo of how to use resurrector.

Fortunately my work has been in different sections of the project, so the only conflicts were in **/.gitignore** and **/setup/steps/install_rclone.sh**, where I forgot to update the archive type, to .zip from .tar.gz, on [line 16](https://github.com/magao-x/MagAOX/blob/dev/setup/steps/install_rclone.sh#L16) in [this Pull Request](https://github.com/magao-x/MagAOX/pull/114) that Joseph merged a week ago.

Even so, this has been a bit of a hot mess and distracted us from the SC work, so we do not plan to do this in the future.

Brian Carcich

### CODE REVIEW

File: setup/steps/install_python.sh [hunk 1]
```diff
@@ -18,18 +18,18 @@ cd /opt/MagAOX/vendor
...
     find /opt/conda -type d -exec sudo chmod g+rwxs {} \;
-    # Set environment variables for conda
+    # Set BASH environment variables for conda
     cat << 'EOF' | tee /etc/profile.d/conda.sh
 if [ -f "/opt/conda/etc/profile.d/conda.sh" ]; then
     . "/opt/conda/etc/profile.d/conda.sh"
     CONDA_CHANGEPS1=false conda activate base
 else
-    \export PATH="/opt/conda/bin:$PATH"
+    \export PATH="/opt/conda/bin:\$PATH"
```
Comment (joseph-long) [inline setup/steps/install_python.sh:32]: This will not work correctly; it adds a literal `$` to the PATH and drops the previous value.

Comment (drbitboy) [reply setup/steps/install_python.sh:32]: I agree.  Now where the heck did that come from?

**Comment by joseph-long** [comment]
I've been working with @kvangorkom on provisioning and ARM support. We'll probably have some conflicts limited to the setup/ subfolder when it comes time to merge. We could do some git gymnastics to split those out from this PR, or just resolve the conflicts later when we're ready to test the resurrector

**Comment by joseph-long** [comment]
@drbitboy I have created a branch called `resurrector` in this repo that is up to date with the changes here. I resolved the merge conflicts in the setup scripts. We want to be able to test the resurrector and retain the ability to switch back and forth for now. We can periodically merge or rebase on dev.

If you have push access here, you can feel free to push directly to magao-x/MagAOX:resurrector; otherwise you can make a PR to that branch for new changes. Thanks!

**Pull request closed without merging**